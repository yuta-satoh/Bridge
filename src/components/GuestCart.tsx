import useSWR, { Fetcher } from "swr";
import Link from "next/link";
import Image from "next/image";
import Recommend from "./Recommend";

// カートに入った時の状態を含めたいのでcartInfoプロパティも追加しています
type Item = {
  id: number,
  name: string,
  description: string,
  genre: string,
  category: string,
  price: number,
  imgurl: string,
  stock: number,
  delete: boolean,
}

const fetcher: Fetcher<Item[], string> = (...args) => fetch(...args).then((res) => res.json());

export default function GuestCart({ guestCart }: { guestCart: string[] }) {
  // 受け取ったゲストカートからクエリパラメータを作成
  const itemQuery = guestCart.reduce((query, cartItem) => query + `,id.eq.${cartItem}`, "").replace(",", "");
  const queryParams = `or=(${itemQuery})`

	// SWRでアイテムを取得
  const { data: cartItemData, error } = useSWR(`/api/getItems?id=${queryParams}`, fetcher)

  // カートにデータがない時の表示
	if (error) return (
    <div className="w-4/5 mx-auto">
      <h2 className="text-3xl font-bold mt-10">カート</h2>
      <div className="block py-40 mx-auto my-10 bg-neutral-100 rounded">
        <h2 className="text-center text-3xl font-bold mb-10">カートに商品がありません</h2>
        <Link href="/">
          <div className="container mx-auto pt-1.5 text-center h-10 w-1/3 border-2 border-neutral-900 bg-white mt-4">
            お買い物はこちらから
          </div>
        </Link>
      </div>
    </div>
  )

	if (!cartItemData) return <div>loading...</div>
  
  // deleteフラグの立っているアイテムを除去
  const filteredItemData = cartItemData.filter((item) => !item.delete)
  
  // 合計金額
  const sumPrice = filteredItemData.reduce((current, item) => current + item.price, 0)

	return (
    <div className="flex mx-auto w-4/5 gap-32">
      <div className="w-3/5 mt-10">
        <div className="flex">
          <h2 className="text-3xl font-bold">カート</h2>
          <span className="mt-2">{`(${filteredItemData.length}点の商品)`}</span>
        </div>
        {filteredItemData.map((item, index) => (
          <div key={index} className="border border-neutral-900 my-2 py-3 px-8 h-52">
            <div className="flex gap-5">
              <Link href={'/'}>
                <Image src={item.imgurl} alt={item.name} width={150} height={150} className="rounded"/>
              </Link>
              <div className="px-3 py-4">
                <Link href={'/'}>
                  <p className="underline mb-1 text-xl">{item.name}</p>
                </Link>
                <p className="text-sm mt-1">{item.description}</p>
                <p className="mt-1 text-lg">¥ {item.price.toLocaleString()}</p>
              </div>
            </div>
            <div className="text-right">
              <button className="text-white bg-neutral-900 border border-neutral-900 rounded px-1">
                削除
              </button>
              {/* 個数を変えたら金額も変更したい（CSRで） */}
              <select
                name="quantity"
                id="cart_quantity"
                className="ml-5 border border-neutral-900 rounded p-1"
              >
                <option value={1}>1</option>
                <option value={2}>2</option>
                <option value={3}>3</option>
                <option value={4}>4</option>
                <option value={5}>5</option>
                <option value={6}>6</option>
                <option value={7}>7</option>
                <option value={8}>8</option>
                <option value={9}>9</option>
                <option value={10}>10</option>
              </select>
            </div>
          </div>  
        ))}

        <Recommend filteredItemData={filteredItemData} />
      </div>
      <div className="w-1/4 h-80 mt-10 p-10 border-2 border-neutral-900 rounded bg-gray-100">
        <p>
          <span>商品合計</span>
          <span className="float-right">¥ {sumPrice.toLocaleString()}</span>
        </p>
        <p>
          <span>消費税</span>
          <span className="float-right">¥ {(sumPrice * 0.1).toLocaleString()}</span>
        </p>
        <div className="border-b border-black">
          <p className="mt-10 text-lg font-bold">
            <span>合計</span>
            <span className="float-right">¥ {(sumPrice + (sumPrice * 0.1)).toLocaleString()}</span>
          </p>
        </div>
        <Link href="/login">
          <div className="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-8">
            <span>ログイン</span>
          </div>
        </Link>
        <Link href="/">
          <div className="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-4">
            <span>お買い物を続ける</span>
          </div>
        </Link>
      </div>
    </div>
  )
}
