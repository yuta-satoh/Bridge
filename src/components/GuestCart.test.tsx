import { render, screen } from "@testing-library/react";
import { Middleware, SWRResponse, SWRConfig } from "swr";
import userEvent from "@testing-library/user-event";

import GuestCart from "./GuestCart";

jest.mock('next/router', () => ({
    useRouter() {
      return {
        replace: jest.fn(),
        push: jest.fn(),
        reload: jest.fn(),
      };
    },
}));

const fetchMockOk = () => {
    return new Promise((resolve) => {
        resolve({
            ok: true,
            status: 200,
            statusText: 'OK',
            json: () => '1'
        })
    })
}

describe('ゲストカートコンポーネント', () => {
    test('ゲストカートが空の状態', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                return {
                    data: [],
                    error: Error(),
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <GuestCart guestCart={[]} reloadStrage={jest.fn()} />
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ゲストカートをローディング中', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                return {
                    data: undefined,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <GuestCart guestCart={[]} reloadStrage={jest.fn()} />
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ゲストカートに中身あり', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        name: "ベルベットアームチェア",
                        description: "ベルベットのお洒落なデザインチェア。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 16000,
                        imgurl: "/images/chair/chair_feminine_1.jpg",
                        stock: 72,
                        delete: false
                    },
                    {
                        id: 2,
                        name: "丸椅子スツール",
                        description: "落ち着いた可愛らしいベロア風スツール。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 10000,
                        imgurl: "/images/chair/chair_feminine_2.jpg",
                        stock: 99,
                        delete: false
                    }, 
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <GuestCart guestCart={[{ itemId: 1, quantity: 1 }, { itemId: 2, quantity: 1 }]} reloadStrage={jest.fn()}/>
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ゲストカートの商品の個数を変更する', async () => {
        global.fetch = jest.fn().mockImplementation(fetchMockOk);
        const user = userEvent.setup()

        // 初期レンダリング用モックデータ
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        name: "ベルベットアームチェア",
                        description: "ベルベットのお洒落なデザインチェア。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 16000,
                        imgurl: "/images/chair/chair_feminine_1.jpg",
                        stock: 72,
                        delete: false
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }

        // 個数変更後のモックデータ
        const changeMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        name: "ベルベットアームチェア",
                        description: "ベルベットのお洒落なデザインチェア。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 16000,
                        imgurl: "/images/chair/chair_feminine_1.jpg",
                        stock: 72,
                        delete: false
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const {asFragment, rerender} = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <GuestCart guestCart={[{ itemId: 1, quantity: 1 }]} reloadStrage={jest.fn()}/>
            </SWRConfig>
        );
        const selectId = screen.getByTestId("select-option");
        await user.selectOptions(selectId, '2');
        // セレクトボックスは変更できたが金額は反映されず
        // 再レンダリングして個数を変更したデータを渡す
        rerender(
            <SWRConfig value={{ use: [changeMiddleware] }}>
                <GuestCart guestCart={[{itemId: 1, quantity: 2}]} reloadStrage={jest.fn()} />
            </SWRConfig>
        )
        expect(asFragment()).toMatchSnapshot();
    })

    test('ゲストカートの商品を削除', async () => {
        global.fetch = jest.fn().mockImplementation(fetchMockOk);
        const user = userEvent.setup();
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        name: "ベルベットアームチェア",
                        description: "ベルベットのお洒落なデザインチェア。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 16000,
                        imgurl: "/images/chair/chair_feminine_1.jpg",
                        stock: 72,
                        delete: false
                    },
                    {
                        id: 2,
                        name: "丸椅子スツール",
                        description: "落ち着いた可愛らしいベロア風スツール。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 10000,
                        imgurl: "/images/chair/chair_feminine_2.jpg",
                        stock: 99,
                        delete: false
                    }, 
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const deleteMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        name: "ベルベットアームチェア",
                        description: "ベルベットのお洒落なデザインチェア。",
                        genre: "フェミニン",
                        category: "椅子",
                        price: 16000,
                        imgurl: "/images/chair/chair_feminine_1.jpg",
                        stock: 72,
                        delete: false
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment, rerender } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <GuestCart guestCart={[{itemId: 1, quantity: 1}, { itemId: 2, quantity: 1 }]} reloadStrage={jest.fn()} />
            </SWRConfig>
        );
        const deleteBtn = screen.getAllByRole('button');
        await user.click(deleteBtn[1]);
        // 再レンダリングして削除後のデータを渡す
        rerender(
            <SWRConfig value={{ use: [deleteMiddleware] }}>
                <GuestCart guestCart={[{itemId: 1, quantity: 1}]} reloadStrage={() => jest.fn()} />
            </SWRConfig>    
        )
        expect(asFragment()).toMatchSnapshot();
    })
})
