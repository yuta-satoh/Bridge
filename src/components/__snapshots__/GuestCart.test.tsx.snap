// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ゲストカートコンポーネント ゲストカートが空の状態 1`] = `
<DocumentFragment>
  <div
    class="w-4/5 mx-auto"
  >
    <h2
      class="text-3xl font-bold mt-10"
    >
      カート
    </h2>
    <div
      class="block py-40 mx-auto my-10 bg-neutral-100 rounded"
    >
      <h2
        class="text-center text-3xl font-bold mb-10"
      >
        カートに商品がありません
      </h2>
      <a
        href="/"
      >
        <div
          class="container mx-auto pt-1.5 text-center h-10 w-1/3 border-2 border-neutral-900 bg-white mt-4"
        >
          お買い物はこちらから
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ゲストカートコンポーネント ゲストカートに中身あり 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (2点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded w-36 h-36"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="jsx-3340056729"
            type="button"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/2"
          >
            <img
              alt="丸椅子スツール"
              class="rounded w-36 h-36"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/2"
            >
              <p
                class="underline mb-1 text-xl"
              >
                丸椅子スツール
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              落ち着いた可愛らしいベロア風スツール。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 10,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="jsx-3340056729"
            type="button"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        class="mt-16 mb-5"
      >
        <h2
          class="text-3xl font-bold"
        >
          こちらもいかがですか？
        </h2>
      </div>
      <div
        class="flex gap-3 mb-10"
      >
        <div
          class="border border-neutral-900 w-1/2 rounded p-3 mx-auto"
        >
          <div
            class="flex gap-5"
          >
            <a
              href="/items/itemlist/1"
            >
              <img
                alt="ベルベットアームチェア"
                class="rounded"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
            </a>
            <div
              class="px-2 py-2"
            >
              <a
                href="/items/itemlist/1"
              >
                <p
                  class="underline mb-1 text-sm"
                >
                  ベルベットアームチェア
                </p>
                <p
                  class="mt-1 mb-5"
                >
                  ¥ 16,000
                </p>
              </a>
              <button
                class="jsx-896052889"
                type="button"
              >
                カートに入れる
              </button>
            </div>
          </div>
        </div>
        <div
          class="border border-neutral-900 w-1/2 rounded p-3 mx-auto"
        >
          <div
            class="flex gap-5"
          >
            <a
              href="/items/itemlist/2"
            >
              <img
                alt="丸椅子スツール"
                class="rounded"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
            </a>
            <div
              class="px-2 py-2"
            >
              <a
                href="/items/itemlist/2"
              >
                <p
                  class="underline mb-1 text-sm"
                >
                  丸椅子スツール
                </p>
                <p
                  class="mt-1 mb-5"
                >
                  ¥ 10,000
                </p>
              </a>
              <button
                class="jsx-896052889"
                type="button"
              >
                カートに入れる
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-1/4 h-80 mt-10 p-10 border-2 border-neutral-900 rounded bg-gray-100"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 26,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 2,600
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 28,600
          </span>
        </p>
      </div>
      <div
        class="flex flex-col items-center gap-5 mt-5"
      >
        <button
          class="jsx-1912866866"
          type="button"
        >
          ログイン
        </button>
        <button
          class="jsx-1912866866"
          type="button"
        >
          お買い物を続ける
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ゲストカートコンポーネント ゲストカートの商品の個数を変更する 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (1点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded w-36 h-36"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="jsx-3340056729"
            type="button"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        class="mt-16 mb-5"
      >
        <h2
          class="text-3xl font-bold"
        >
          こちらもいかがですか？
        </h2>
      </div>
      <div
        class="flex gap-3 mb-10"
      >
        <div
          class="border border-neutral-900 w-1/2 rounded p-3 mx-auto"
        >
          <div
            class="flex gap-5"
          >
            <a
              href="/items/itemlist/1"
            >
              <img
                alt="ベルベットアームチェア"
                class="rounded"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
            </a>
            <div
              class="px-2 py-2"
            >
              <a
                href="/items/itemlist/1"
              >
                <p
                  class="underline mb-1 text-sm"
                >
                  ベルベットアームチェア
                </p>
                <p
                  class="mt-1 mb-5"
                >
                  ¥ 16,000
                </p>
              </a>
              <button
                class="jsx-896052889"
                type="button"
              >
                カートに入れる
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-1/4 h-80 mt-10 p-10 border-2 border-neutral-900 rounded bg-gray-100"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 32,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 3,200
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 35,200
          </span>
        </p>
      </div>
      <div
        class="flex flex-col items-center gap-5 mt-5"
      >
        <button
          class="jsx-1912866866"
          type="button"
        >
          ログイン
        </button>
        <button
          class="jsx-1912866866"
          type="button"
        >
          お買い物を続ける
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ゲストカートコンポーネント ゲストカートの商品を削除 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (1点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded w-36 h-36"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="jsx-3340056729"
            type="button"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        class="mt-16 mb-5"
      >
        <h2
          class="text-3xl font-bold"
        >
          こちらもいかがですか？
        </h2>
      </div>
      <div
        class="flex gap-3 mb-10"
      >
        <div
          class="border border-neutral-900 w-1/2 rounded p-3 mx-auto"
        >
          <div
            class="flex gap-5"
          >
            <a
              href="/items/itemlist/1"
            >
              <img
                alt="ベルベットアームチェア"
                class="rounded"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
            </a>
            <div
              class="px-2 py-2"
            >
              <a
                href="/items/itemlist/1"
              >
                <p
                  class="underline mb-1 text-sm"
                >
                  ベルベットアームチェア
                </p>
                <p
                  class="mt-1 mb-5"
                >
                  ¥ 16,000
                </p>
              </a>
              <button
                class="jsx-896052889"
                type="button"
              >
                カートに入れる
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-1/4 h-80 mt-10 p-10 border-2 border-neutral-900 rounded bg-gray-100"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 16,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 1,600
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 17,600
          </span>
        </p>
      </div>
      <div
        class="flex flex-col items-center gap-5 mt-5"
      >
        <button
          class="jsx-1912866866"
          type="button"
        >
          ログイン
        </button>
        <button
          class="jsx-1912866866"
          type="button"
        >
          お買い物を続ける
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ゲストカートコンポーネント ゲストカートをローディング中 1`] = `
<DocumentFragment>
  <div
    class="jsx-1008565373 spinner mx-auto"
  >
    <span
      class="jsx-1008565373"
    >
      L
    </span>
    <span
      class="jsx-1008565373"
    >
      O
    </span>
    <span
      class="jsx-1008565373"
    >
      A
    </span>
    <span
      class="jsx-1008565373"
    >
      D
    </span>
    <span
      class="jsx-1008565373"
    >
      I
    </span>
    <span
      class="jsx-1008565373"
    >
      N
    </span>
    <span
      class="jsx-1008565373"
    >
      G
    </span>
  </div>
</DocumentFragment>
`;
